verification/verify.go

```diff
@@ -153,3 +160,75 @@ func IsRetryable(err error) bool {
 	_, retryable := err.(Retryable)
 	return retryable
 }
+
+func verifyPixelParams(params *Params) error {
+	// check if the node is on-chain mode
+	if params != nil &&
+		params.Orchestrator != nil &&
+		params.Orchestrator.TicketParams != nil {
+
+		var err error
+		for i := 0; err == nil && i < len(params.Results.Segments); i++ {
+			rendition := []byte{}
+			if i < len(params.Renditions) {
+				rendition = params.Renditions[i]
+			}
+			if err := verifyPixels(
+				params.Results.Segments[i].Url,
+				rendition,
+				params.Results.Segments[i].Pixels); err != nil {
+
+				glog.Error(err)
+				return ErrPixelMismatch
+			}
+		}
+		return nil
+
+	}
+	return ErrPixelsAbsent
+}
+
+func verifyPixels(fname string, data []byte, reportedPixels int64) error {
+	uri, err := url.ParseRequestURI(fname)
+	// If the filename is a relative URI and the broadcaster is using local memory storage
+	// write the data to a temp file
+	if err == nil && !uri.IsAbs() && data != nil {
+		tempfile, err := ioutil.TempFile("", common.RandName())
+		if err != nil {
+			return fmt.Errorf("error creating temp file for pixels verification: %v", err)
```

Comment:
Even though this bit was copy-pasted from the old code... since I think we basically require Go 1.13 now, maybe we could use the %w modifier to make the original error type introspectable? See https://blog.golang.org/go1.13-errors

Reply:
thanks :-). done in e96c310d6d949b8dd76cc857f3b1b0949a8bc46a